skyportal/handlers/api/observation_plan.py

```diff
@@ -625,16 +700,64 @@ def get(self, observation_plan_request_id=None):
 
-            observation_plan_requests = (
-                session.scalars(
-                    ObservationPlanRequest.select(
-                        session.user_or_token, options=options
-                    )
+            observation_plan_requests = ObservationPlanRequest.select(
+                session.user_or_token, options=options
+            )
+
+            if start_date:
+                start_date = str(arrow.get(start_date.strip()).datetime)
+                observation_plan_requests = observation_plan_requests.where(
+                    ObservationPlanRequest.created_at >= start_date
                 )
-                .unique()
-                .all()
+            if end_date:
+                end_date = str(arrow.get(end_date.strip()).datetime)
+                observation_plan_requests = observation_plan_requests.where(
+                    ObservationPlanRequest.created_at <= end_date
+                )
+            if dateobs:
+                gcn_event_query = GcnEvent.select(self.current_user).where(
+                    GcnEvent.dateobs == dateobs
+                )
+                gcn_event_subquery = gcn_event_query.subquery()
+                observation_plan_requests = observation_plan_requests.join(
+                    gcn_event_subquery,
+                    ObservationPlanRequest.gcnevent_id == gcn_event_subquery.c.id,
+                )
+            if instrumentID:
+                # allocation query required as only way to reach
+                # instrument_id is through allocation (as requests
+                # are associated to allocations, not instruments)
+                allocation_query = Allocation.select(self.current_user).where(
+                    Allocation.instrument_id == instrumentID
+                )
+                allocation_subquery = allocation_query.subquery()
+                observation_plan_requests = observation_plan_requests.join(
+                    allocation_subquery,
+                    ObservationPlanRequest.allocation_id == allocation_subquery.c.id,
+                )
+            if status:
+                observation_plan_requests = observation_plan_requests.where(
+                    ObservationPlanRequest.status.contains(status.strip())
+                )
+
+            count_stmt = sa.select(func.count()).select_from(observation_plan_requests)
+            total_matches = session.execute(count_stmt).scalar()
+            if n_per_page is not None:
+                observation_plan_requests = (
+                    observation_plan_requests.distinct()
+                    .limit(n_per_page)
+                    .offset((page_number - 1) * n_per_page)
+                )
+            observation_plan_requests = (
+                session.scalars(observation_plan_requests).unique().all()
             )
-            return self.success(data=observation_plan_requests)
+
+            info = {}
+            info["observation_plan_requests"] = [
```

Comment:
Maybe change the key from `observation_plan_requests` to `request` just to make it easier for the users?

Reply:
great idea.